src/annotations/api/index.test.ts

```diff
@@ -1,52 +1,189 @@
+
+    // Handles spaces
+    expect(qs.indexOf(' ')).toEqual(-1)
+    expect(qs.includes('default%20stream')).toBeTruthy()
+
+    // Only uses ? symbol in beginning, following qs params use &
+    expect(qs.lastIndexOf('?')).toEqual(0)
+    expect(qs.split('').filter(a => a === '&').length).toEqual(3)
+
+    // Handles encoding of object params for stickers
+    expect(qs.includes('&stickerIncludes[foo]=bar')).toBeTruthy()
+  })
+})
 describe('annotations api calls', () => {
   describe('POST - write annotations api calls', () => {
     it('can write an annotation successfully and return the array of annotations with camelcased response properties', async () => {
       const data = [
         {
           summary: 'GO PACK GO',
-          'start-time': Date.now(),
-          'end-time': Date.now(),
+          start: Date.now().toString(),
+          end: Date.now().toString(),
           stream: 'Lambeau Field',
         },
       ]
 
-      const transformedData = [
+      mocked(axios.post).mockImplementationOnce(() =>
+        Promise.resolve({status: 200, data})
+      )
+
+      const response = await writeAnnotation(data)
+
+      expect(axios.post).toHaveBeenCalledTimes(1)
+
+      expect(response).toEqual(data)
+    })
+
+    it('handles an error and returns the error message', async () => {
+      const data = [
         {
           summary: 'GO PACK GO',
-          startTime: Date.now(),
-          endTime: Date.now(),
+          start: Date.now().toString(),
+          end: Date.now().toString(),
           stream: 'Lambeau Field',
         },
       ]
+      const message = 'OOPS YOU DONE MESSED UP SON'
+
       mocked(axios.post).mockImplementationOnce(() =>
-        Promise.resolve({status: 200, data})
+        Promise.reject(new Error(message))
       )
 
-      const response = await writeAnnotation([{summary: 'Hey there'}])
+      await expect(writeAnnotation(data)).rejects.toThrow(message)
+    })
+  })
+
+  describe('GET - retrieve annotations api calls', () => {
+    const data = [
+      {
+        stream: 'Lambeau Field',
+        annotations: [
+          {
+            start: Date.now().toString(),
+            end: Date.now().toString(),
+            summary: 'Go Pack Go',
+          },
+        ],
+      },
+      {
+        stream: 'default',
+        annotations: [
+          {
+            start: '2020-01-12T23:13:47Z',
+            end: Date.now().toString(),
+            summary: 'blah blah',
+          },
+        ],
+      },
+    ]
 
-      expect(axios.post).toHaveBeenCalledTimes(1)
+    it('retrieves annotations and returns them categorized by annotation stream', async () => {
+      mocked(axios.get).mockImplementationOnce(() => Promise.resolve({data}))
+      const response = await getAnnotation({
+        start: Date.now().toString(),
+        end: Date.now().toString(),
+        stream: 'Lambeau Field',
+      })
 
-      expect(response).toEqual(transformedData)
+      expect(response).toEqual(data)
```

Comment:
if it's requesting the "Lambeau field" annotations, it gets the default too?
(that is what the code says)

that seems counter-intuitive to me.

Reply:
No. If it requests Lambeau Field it will only get Lambeau Field annotations. I'll update the test to reflect that, but its because of the value in the mock needing to be changed, not because it doesn't filter by stream